frontend/webservice/vrf.cgi

```diff
@@ -381,50 +277,147 @@ sub provision_vrf{
+                #found the same endpoint..
+                $found = 1;
+                #remove the endpoint from the endpoint list
+                splice(@{$model->{'endpoints'}},$i,1);
+
+                my @new_peers;
+                #now compare the Peers
+                foreach my $peer (@{$ep->peers()}){
+                    my $peer_found = 0;
+                    for( my $j =0 ; $j<= $#{$new_ep->{'peerings'}}; $j++){
+                        my $new_peer = $new_ep->{'peerings'}->[$j];
+                        next if($new_peer->{'peer_ip'} ne $peer->peer_ip());
+                        next if($new_peer->{'local_ip'} ne $peer->local_ip());
+                        next if($new_peer->{'asn'} ne $peer->peer_asn());
+                        next if($new_peer->{'key'} ne $peer->md5_key());
+                        $peer_found = 1;
+                
+                        #remove the peering from the new_ep
+                        splice(@{$new_ep->{'peerings'}},$j,1);
+                        last;                                
+                    }
+                    if($peer_found){
+                        push(@new_peers, $peer);                        
+                    }
+                }
+                
+                foreach my $model_peer (@{$new_ep->{'peerings'}}){
+                    my $peer = OESS::Peer->new( model => $model_peer, db => $db );
+                    push(@new_peers, $peer);
+                }
+                
+                $ep->peers(\@new_peers);
+
+            }
+        }
+        if($found){
+            #we found the endpoint and updated any peerings
+            push(@new_endpoints, $ep);
+            #there should be no need to do CLOUD stuff here! (unless we added a second peering)
+        }else{
+            #if it isn't found we don't add it to new_endpoints
+            #is this a cloud connection?
+            if($ep->{'cloud_account_id'}){
+                push(@cloud_dels, $ep);
+            }
+        }
+    }
+    
+    if($#{$model->{'endpoints'}} >= 0){
+        foreach my $model_ep (@{$model->{'endpoints'}}){
+            #create an Endpoint object!
+            my $ep = OESS::Endpoint->new( db => $db, model => $model_ep, type => 'vrf');
+            push(@new_endpoints, $ep);
+            if(defined($ep->{'cloud_account_id'})){
+                push(@cloud_adds, $ep);
+            }
+        }
     }
 
-    my $ok = $vrf->create();
-    if (!$ok) {
-        $method->set_error('error creating VRF: ' . $vrf->error());
-        return;
+    $vrf->endpoints(\@new_endpoints);
+
+    #our VRF object has the endpoints... now make any cloud changes (if necessary)
+    if(!$skip_cloud_provisioning){
+        eval {
+            OESS::Cloud::cleanup_endpoints(\@cloud_dels);
+
+            my $setup_endpoints = OESS::Cloud::setup_endpoints($vrf->name, \@cloud_adds);
```

Comment:
`OESS::Cloud::setup_endpoints` configures cloud services for any interface in `$cloud_adds` with a configured cloud interconnect id. Once complete, any information related to new cloud services is recorded in the resulting endpoint list (the cloud_connection_id being most important). The resulting endpoint list should be used to update the VRF's endpoints.